fix: unified search provider id

SebastianKrupinski opened the pull request:
## Issue

Unified Search not searching dav properly

![Screenshot 2025-01-29 154244](https://github.com/user-attachments/assets/a8eb201b-c160-45a8-8cc7-11ec37a003ac)

Unified search backend use the provider id to match providers but the front end was using the app id.

This works fine for most search providers as the app id and provider id are the same in most cases (e.g files app; app id is 'files', search provider id is 'files') but in other cases like dav this breaks (e.g. dav app; app id is 'dav' but search provider is 'contacts or events')

Also using the app id is not a good choice as one app can have multiple providers registered (e.g. dav has Contacts, Events and Tasks search) 

## Resolution

Modify front end to use provider id instead of app id

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

SebastianKrupinski committed:
fix: unified search provider id

Signed-off-by: SebastianKrupinski <krupinskis05@gmail.com>
Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

SebastianKrupinski commented:
/compile amend

nfebe commented on core/src/components/UnifiedSearch/UnifiedSearchModal.vue:
```diff
@@ -374,7 +374,7 @@ export default defineComponent({
 			const searchProvider = (provider, filters) => {
 				const params = {
-					type: provider.appId,
+					type: provider.id,
```
We have to do more than this, it's not clear that the providers don't always match with the appIds

SebastianKrupinski replied:
Each app can have more then one provider, therefore the provider id is the correct way. The provider id can be customized, for each provider

nfebe replied:
Yeah, however it's possible to have multiple providers with the same ID and this is a problem.

nfebe reviewed (changes requested):
I would look into this asap

SebastianKrupinski commented:
@nfebe your commit broker this again.

![image](https://github.com/user-attachments/assets/3f12d4bd-3f5e-4f64-a645-9d3c9e00f2dd)

You can not just make up your own provider id, the id is being sent as 'dav-contacts' which is not valid. 

What issue are you trying to solve maybe I can help you find a proper solution?

nfebe commented:
> You can not just make up your own provider id, the id is being sent as 'dav-contacts' which is not valid.

Sorry, I converted it to WIP, it's not supposed to use that there a corresponding frontend change which I am adjusting.

nfebe committed:
fix: Use `searchFrom` property for client side plugins

The client-side plugin `in-folder` uses the `files` provider, this makes it
overlap with the main files provider itself.

This change follows eecda06f1ad2832cc2a7b31d646458f730c8412a after it was discovered
that some apps/providers like `dav` use providers from another app like `contacts`

Signed-off-by: nfebe <fenn25.fn@gmail.com>
Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

nfebe commented:
> Also using the app id is not a good choice as one app can have multiple providers registered (e.g. dav has Contacts, Events and Tasks search)

Sure, I added a `searchFrom` property to in `in-folder` search plugin we could use this in the backend, my attempted backend changes where reverted as it would need much more work.

I have added a commit that restores full function with apps like dav while still having in-folder search working as expected.

-----

TODO: 

- [ ] Ensure "load more results" removes other filters

SebastianKrupinski commented:
/compile amend

provokateurin reviewed (approved):
Reminds me of https://github.com/nextcloud/server/pull/47523 :disappointed:
I really dislike when this kind of mess comes to haunt us back way later.
It works almost fine because in most cases it works, but we don't test good enough for "edge cases" (which aren't really edge cases tbh because it's just standard behavior).

nfebe commented:
/backport to stable30
